### DISCUSSION TITLE
video export corruption

### DISCUSSION THREAD

**Discussion by MyraBaba** [obss/sahi#634] [Q&A] [answered]
Hi,

I tested a video with : 

> sahi predict --model_path /home/alp2080/Projects/yolov5/yolov5s.pt --model_type yolov5 --source /home/alp2080/DATA/Videos/hiv00038.mp4

and has warning:

> [hevc @ 0x59f8300] Could not find ref with POC 48
08/27/2022 18:04:47 - WARNING - sahi.slicing -   Starting from version `0.10.2`, `auto_slice_resolution=True` is introduced as the default behavior for determining slice height and width automatically calculated by the image properties (resolution, aspect ratio and orientation).
Starting from version `0.10.2`, `auto_slice_resolution=True` is introduced as the default behavior for determining slice height and width automatically calculated by the image properties (resolution, aspect ratio and orientation).
Performing prediction on 24 number of slices.
Prediction time is: 329.07 ms
Performing inference on video frames:  33%|█████████▋                   | 4166/12525 [29:12<57:45,  2.41it/s][hevc @ 0x59f8300] Could not find ref with POC 49
08/27/2022 18:04:47 - WARNING - sahi.slicing -   Starting from version `0.10.2`, `auto_slice_resolution=True` is introduced as the default behavior for determining slice height and width automatically calculated by the image properties (resolution, aspect ratio and orientation).
Starting from version `0.10.2`, `auto_slice_resolution=True` is introduced as the default behavior for determining slice height and width automatically calculated by the image properties (resolution, aspect ratio and orientation).
Performing prediction on 24 number of slices.
Prediction time is: 326.92 ms
Performing inference on video frames:  33%|█████████▋                   | 4167/12525 [29:12<57:45,  2.41it/s][hevc @ 0x59f8300] Could not find ref with POC 50 

resuted video few frame good few frame corrupted as below : 
![sahi](https://user-images.githubusercontent.com/17505439/187036158-37cb0e9b-b751-4146-a2b7-c3e958bc8c05.jpg)

**Comment by fcakyon** [comment]
Can you please provide the video file, model type, model file and config file so that we can debug the issue?

**Comment by MyraBaba** [comment]
cant share publicly due to GDPR

**Comment by fcakyon** [comment, marked as answer]
It is most probably a hevc codec/opencv/ffmpeg related issue rather than SAHI. Check respective repos for a solution.

**Comment by MyraBaba** [comment]
this one working without problem and output video is ok. 

https://github.com/WongKinYiu/yolov7

yor video writer could be use theirs

**Comment by fcakyon** [comment]
What is your file format/extension? What is the video FPS? Can you share these?

**Comment by fcakyon** [comment]
[Our video writer](https://github.com/obss/sahi/blob/072ea62bc14646738ec95437e038f88690a0aa4c/sahi/utils/cv.py#L307) and [yolov7 video writer](https://github.com/WongKinYiu/yolov7/blob/44d8ab41780e24eba563b6794371f29db0902271/detect.py#L155) work exactly same.

**Comment by MyraBaba** [comment]
@fcakyon  I can share video with you from email not publicly. How can I reach you

**Comment by MyraBaba** [comment]
I corrected the video with ffmpeg. muxing problem. But interestingly same video has good output with yolov7..

> sahi predict --model_path /home/xxx/Projects/yolov5/yolov5s.pt --model_type yolov5 --source hixx_u.mp4

`WARNING - sahi.slicing -   Starting from version `0.10.2`, `auto_slice_resolution=True` is introduced as the default behavior for determining slice height and width automatically calculated by the image properties (resolution, aspect ratio and orientation).
Starting from version `0.10.2`, `auto_slice_resolution=True` is introduced as the default behavior for determining slice height and width automatically calculated by the image properties (resolution, aspect ratio and orientation).
Performing prediction on 24 number of slices.
Prediction time is: 322.94 ms
Performing inference on video frames:   1%|▋                                                                      | 118/12534 [01:15<1:48:27,  1.91it/s]Performing inference on video frames:   1%|▋                                                                      | 118/12534 [01:15<2:12:27,  1.56it/s]

1,6 g per second  witg 2080 RTX ti . its very slow.  
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 2560x1440, 10381 kb/s, 25 fps, 25 tbr, 12800 tbn, 50 tbc (default)

How we can make it faster ?

**Comment by MyraBaba** [comment]
@fcakyon  is there any convenient way to export boinding boxes with confidence number per frame as csv ? xxwh format

**Comment by fcakyon** [comment]
We currently support pickle export, add ‘—export_pickle’ argument to ‘sahi predict’ cli command.

**Comment by fcakyon** [comment]
You can message me over linkedin btw: https://www.linkedin.com/in/fcakyon

**Comment by fcakyon** [comment]
To make it faster, you can increate the slice sizes, increase the slice overlap ratios or decrease the model image size.
